Modules/api_handler.py

```diff
@@ -80,3 +136,74 @@ def __init__(self, hostname: str, token: str=None, tls=False):
             await self._connection.send(data)
         else:
             await self._connection.send(json.dumps(data))
+
+    def construct_request(self, endpoint: (str, str), root_params: {} = {}, meta_params: {} = {}) -> str:
+        """ Constructs the JSON to be passed along the WebSocket channel from the given parameters
+            root_params has the power to overwrite the endpoint!
+            Args:
+                endpoint (str, str): Api Action to execute [example: ("rescues","search")]
+                root_params: Key-value pairs of parameters for the request, these will be processed by the server
+                meta_params: Key-value pairs of parameters that will be included in the "meta" parameter of the request. 
+                    These will not be processed by the server.
+            Returns: 
+                JSON representation of thr request that can be directly passed to send_raw()
+        """
+        request_id = uuid4()
+        while request_id in [*self._waiting_requests, *self._request_responses]: request_id = uuid4()
+        meta_params = {**meta_params, **{"request_id":request_id}}
```

Comment:
This is unnecessarily complicated.
```python
while request_id in self._waiting_requests or request_id in self._request_responses.keys():
    request_id = uuid4()
meta_params["request_id"] = request_id
```
would be a more transparent version of the same thing.